include/net/buf.h

```diff
@@ -468,6 +469,8 @@ struct net_buf {
 
 		/** Fragments associated with this buffer. */
 		struct net_buf *frags;
+
+		TAILQ_ENTRY(net_buf) q;
```

Comment:
The idea with this initial union in `struct net_buf` is that each member should be at most 4 bytes. A quick look-up of TAILQ_ENTRY shows that it expands to a 8-byte struct, which is not suitable here - it penalises every single net_buf in the system whether this feature is required or not (think e.g. of a Bluetooth-only application that doesn't have an IP stack at all). So it'd be good to find another way to accomplish what you need, perhaps using `buf->user_data`.

Reply:
@jhedberg 

Sharing this entry in a union with net_buf->user_data, will lead to the same result, i.e. +4 bytes.

The idea here was to simply double link net bufs, so this could be flexible enough for the TCP window functionality.

If the space is so tight, then something else should be thought:

-	Single-linked queue.
-	Separate data structure.

Reply:
@ozhuraki yep, I realized that user_data would have the same issue shortly after suggesting it. I think the best option might be if you can limit the need to 4 bytes, i.e. go with a singly-linked queue.